osu.Framework/Platform/SDL2DesktopWindow.cs

```
@@ -90,6 +90,24 @@ public Point Position
             }
         }
 
+        private bool resizable;
+
+        /// <summary>
+        /// Returns or sets whether the window is resizable or not.
+        /// </summary>
+        public bool Resizable
```

[As stated in the SDL docs](https://wiki.libsdl.org/SDL_SetWindowResizable), setting this is invalid if fullscreen:

> Note: You can't change the resizable state of a fullscreen window.

This will therefore silently fail (sets the field successfully, but then doesn't actually do anything on the SDL side and reverts to the actual value on the next `updateWindowSpecifics()`). I'm not sure it should be allowed to call this without some kind of error in that case. Either that, or it should be ensured that any potential sets of this property aren't lost (and the appropriate commands are executed the next time the window changes state to windowed).

---

I can just set the default value for this as `true` since its guaranteed that we will always have a resizable window on startup. Unless we choose to expose this to the user as a framework config variable.

https://github.com/ppy/osu-framework/blob/4b4dc843aff09dc71b92dbb4cc6606a74eafb94f/osu.Framework/Platform/SDL2DesktopWindow.cs#L411-L417

To ensure that `Resizable` is still the correct value, I can just call `SDL.SDL_SetWindowResizable` again after the window state has been changed to `Normal`.

---

> I can just set the default value for this as true since its guaranteed that we will always have a resizable window on startup.

That's kind of unrelated to what I was asking about...?

> To ensure that `Resizable` is still the correct value, I can just call `SDL.SDL_SetWindowResizable` again after the window state has been changed to `Normal`.

That would be one of the ways I proposed to fix this, yes. Note that this would mean that the semantics of `Resizable` we expose would now differ from SDL's (in fullscreen, SDL says that the window is *not* resizable - so it's a byproduct of `SetWindowResizable` calls and the window mode, while ours would be an internal state variable, that doesn't change value with window mode changes, but is only valid in windowed). It should be documented as such.

---

> That's kind of unrelated to what I was asking about...?

Pardon me. I was referring to removing this:

https://github.com/ppy/osu-framework/blob/4b4dc843aff09dc71b92dbb4cc6606a74eafb94f/osu.Framework/Platform/SDL2DesktopWindow.cs#L993

Because inside `SDLDesktopWindow.Create()`, one of the flags set when creating the window is `SDL_WindowFlags.SDL_WINDOW_RESIZABLE`. From that, I can naively assume that we have a resizable window from the start.

In other things, I have updated the xmldocs for the property to note this behavior.